File: remove-element/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int l = 0, r = nums.length;
        while (l < r) {
            if (nums[l] == val) {
                int temp = nums[l];
                nums[l] = nums[r-1];
                nums[r-1] = temp;
                r--;
            } else {
                l++;
            }
        }
        return r;
    }
}

File: remove-element/remove-element.php
class Solution {

    /**
     * @param Integer[] $nums
     * @param Integer $val
     * @return Integer
     */
    function removeElement(&$nums, $val) {
        $k = 0;
        for ($i = 0; $i < count($nums); $i++) {
            if ($nums[$i] != $val) {
                $temp = $nums[$k];
                $nums[$k] = $nums[$i];
                $nums[$i] = $temp;
                $k++;
            }
        }
        return $k;
    }
}